src/ModelObjects/representations.rs

```diff
@@ -24,6 +24,49 @@ pub enum BoolExpression {
 }
 
 impl BoolExpression {
+    pub fn encode_expr(&self) -> String {
+        match self {
+            BoolExpression::AndOp(left, right) => [
+                left.encode_expr(),
+                String::from(" && "),
+                right.encode_expr(),
+            ]
+            .concat(),
+            BoolExpression::OrOp(left, right) => [
+                left.encode_expr(),
+                String::from(" || "),
+                right.encode_expr(),
+            ]
+            .concat(),
+            BoolExpression::LessEQ(left, right) => {
+                [left.encode_expr(), String::from("<="), right.encode_expr()].concat()
+            }
+            BoolExpression::GreatEQ(left, right) => {
+                [left.encode_expr(), String::from(">="), right.encode_expr()].concat()
+            }
+            BoolExpression::LessT(left, right) => {
+                [left.encode_expr(), String::from("<"), right.encode_expr()].concat()
+            }
+            BoolExpression::GreatT(left, right) => {
+                [left.encode_expr(), String::from(">"), right.encode_expr()].concat()
+            }
+            BoolExpression::EQ(left, right) => {
+                [left.encode_expr(), String::from("=="), right.encode_expr()].concat()
+            }
+            BoolExpression::Parentheses(expr) => {
+                [String::from("("), expr.encode_expr(), String::from(")")].concat()
+            }
+            BoolExpression::Clock(clock) => [String::from("??")].concat(),
```

Comment:
is it possible to pass the component or its declarations as a parameter here with Serde? If so can't we look up the name of the clock here